packages/nuxt/src/core/app.ts

```diff
@@ -19,11 +20,12 @@ export function createApp (nuxt: Nuxt, options: Partial<NuxtApp> = {}): NuxtApp
 }
 
 export async function generateApp (nuxt: Nuxt, app: NuxtApp, options: { filter?: (template: ResolvedNuxtTemplate<any>) => boolean } = {}) {
+  const nuxtOptions=nuxt.options
```

Comment:
There are no high-level performance improvements by doing this. The non-destructuring pattern is used across our codebases to make it easier to trace back to the source of each variable and increase code-portability/consistency by sharing the same context `nuxt` everywhere.

---

BTW nice to read the article: https://mathiasbynens.be/notes/shapes-ics (in short if in theory say we wanted to call `generateApp` hundreds of times, js engines such as v8 automatically optimize property access using inline-caches)

Reply:
Thanks for the review! I will run a few benchmarks if you don't mind to see whether or not this is an improvement or not.

Reply:
I also think that despite the inline-caching there's a limit to what it can do, seeing as there are over 15 duplicate indexes, from my benchmarks it seems like it does decrease the performance.